templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml

```
@@ -20,89 +20,140 @@ contents:
+
+            # Pull container image outside of the timeout in a background process to workaround possible
+            # image corruption
+            # https://github.com/containers/podman/issues/14003
+            if ! /usr/bin/podman image exists {{ .Images.baremetalRuntimeCfgImage }}; then
+                # Multiple podman pulls may occur in parallel wasting bandwidth therefore this
+                # needs to be removed when podman image corruption issue is resolved.
+                # Send to background to avoid SIGTERM from timeout which may cause image corruption issue during image pull
+                pull_baremetal_runtime_cfg_image &
+                # Wait until image is available
+                >&2 echo "NM resolv-prepender: Waiting for baremetal runtime cfg image to be available"
+                until /usr/bin/podman image exists {{ .Images.baremetalRuntimeCfgImage }}; do
+                    sleep 1
+                done
+                >&2 echo "NM resolv-prepender: Baremetal runtime cfg image is available"
+            fi
 
-        NAMESERVER_IP=$(timeout 20s /usr/bin/podman run --rm \
-            --authfile /var/lib/kubelet/config.json \
-            --net=host \
-            {{ .Images.baremetalRuntimeCfgImage }} \
-            node-ip \
-            show \
-            --retry-on-failure \
-            {{range onPremPlatformAPIServerInternalIPs . }}"{{.}}" {{end}} \
-            {{range onPremPlatformIngressIPs . }}"{{.}}" {{end}})
-        DOMAINS="${IP4_DOMAINS} ${IP6_DOMAINS} {{.DNS.Spec.BaseDomain}}"
-        if [[ -n "$NAMESERVER_IP" ]]; then
-            if systemctl -q is-enabled systemd-resolved; then
-                >&2 echo "NM resolv-prepender: Setting up systemd-resolved for OKD domain and local IP"
-                if [[ ! -f /etc/systemd/resolved.conf.d/60-kni.conf ]]; then
-                    >&2 echo "NM resolv-prepender: Creating /etc/systemd/resolved.conf.d/60-kni.conf"
-                    mkdir -p /etc/systemd/resolved.conf.d
-                    echo "[Resolve]" > /etc/systemd/resolved.conf.d/60-kni.conf
-                    echo "DNS=$NAMESERVER_IP" >> /etc/systemd/resolved.conf.d/60-kni.conf
-                    echo "Domains=$DOMAINS" >> /etc/systemd/resolved.conf.d/60-kni.conf
-                    if systemctl -q is-active systemd-resolved; then
-                        >&2 echo "NM resolv-prepender: restarting systemd-resolved"
-                        systemctl restart systemd-resolved
+            # Image must exist before reaching this point otherwise timeout may cause image corruption
+            NAMESERVER_IP=$(/usr/bin/podman run --rm \
+                --authfile /var/lib/kubelet/config.json \
+                --net=host \
+                {{ .Images.baremetalRuntimeCfgImage }} \
+                node-ip \
+                show \
+                --retry-on-failure \
+                "{{ onPremPlatformAPIServerInternalIP . }}" \
+                "{{ onPremPlatformIngressIP . }}")
```

Was this change intentional (you're partially reverting [PR 3269](https://github.com/openshift/machine-config-operator/commit/87a51419c1662177947d6a4b3f5f42cad13d0a66#diff-255f8a4599166f31961853ea8626f969ca4231c55aacbc20a5bb3ceb640f911d))? IMO it should be change it back to:
```suggestion
                {{range onPremPlatformAPIServerInternalIPs . }}"{{.}}" {{end}} \
                {{range onPremPlatformIngressIPs . }}"{{.}}" {{end}})
```

---

No. I have no idea how that got removed. Ill review all the changes again.